Source/FieldSolver/FiniteDifferenceSolver/ApplySilverMuellerBoundary.cpp

```diff
@@ -0,0 +1,157 @@
+    amrex::Box domain_box,
+    amrex::Real const dt ) {
+
+#ifdef WARPX_DIM_RZ
+    amrex::Abort("The Silver-Mueller boundary conditions cannot be used in RZ geometry.");
+#else
+
+    // Ensure that we are using the Yee solver
+    if (m_fdtd_algo != MaxwellSolverAlgo::Yee) {
+        amrex::Abort("The Silver-Mueller boundary conditions can only be used with the Yee solver.");
+    }
+
+    // Ensure that we are using the cells the domain
+    domain_box.enclosedCells();
+
+    // Calculate relevant coefficients
+    amrex::Real const cdt_over_dx = PhysConst::c*dt*m_stencil_coefs_x[0];
+    amrex::Real const coef1_x = (1._rt - cdt_over_dx)/(1._rt + cdt_over_dx);
+    amrex::Real const coef2_x = 2._rt*cdt_over_dx/(1._rt + cdt_over_dx) / PhysConst::c;
+#ifdef WARPX_DIM_3D
+    amrex::Real const cdt_over_dy = PhysConst::c*dt*m_stencil_coefs_y[0];
+    amrex::Real const coef1_y = (1._rt - cdt_over_dy)/(1._rt + cdt_over_dx);
+    amrex::Real const coef2_y = 2._rt*cdt_over_dy/(1._rt + cdt_over_dy) / PhysConst::c;
+#endif
+    amrex::Real const cdt_over_dz = PhysConst::c*dt*m_stencil_coefs_z[0];
+    amrex::Real const coef1_z = (1._rt - cdt_over_dz)/(1._rt + cdt_over_dx);
+    amrex::Real const coef2_z = 2._rt*cdt_over_dz/(1._rt + cdt_over_dz) / PhysConst::c;
+
+    // Loop through the grids, and over the tiles within each grid
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+    // tiling is usually set by TilingIfNotGPU()
+    // but here, we set it to false because of potential race condition,
+    // since we grow the tiles by one guard cell after creating them.
+    for ( MFIter mfi(*Efield[0], false); mfi.isValid(); ++mfi ) {
+
+        // Extract field data for this grid/tile
+        Array4<Real> const& Ex = Efield[0]->array(mfi);
+        Array4<Real> const& Ey = Efield[1]->array(mfi);
+        Array4<Real> const& Ez = Efield[2]->array(mfi);
+        Array4<Real> const& Bx = Bfield[0]->array(mfi);
+        Array4<Real> const& By = Bfield[1]->array(mfi);
+        Array4<Real> const& Bz = Bfield[2]->array(mfi);
+
+        // Extract the tileboxes for which to loop
+        Box tbx  = mfi.tilebox(Bfield[0]->ixType().toIntVect());
+        Box tby  = mfi.tilebox(Bfield[1]->ixType().toIntVect());
+        Box tbz  = mfi.tilebox(Bfield[2]->ixType().toIntVect());
+
+        // We will modify the first (i.e. innermost) guard cell
+        // (if it is outside of the physical domain)
+        // Thus, the tileboxes here are grown by 1 guard cell
+        tbx.grow(1);
+        tby.grow(1);
+        tbz.grow(1);
+
+        // Apply Boundary condition to Bx
+        amrex::ParallelFor(tbx,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k){
+
+#ifdef WARPX_DIM_3D
+                // At the +y boundary (innermost guard cell)
+                if ( j==domain_box.bigEnd(1)+1 )
+                    Bx(i,j,k) = coef1_y * Bx(i,j,k) + coef2_y * Ez(i,j,k);
```

Comment:
The comment earlier says update Efield at the boundary, but, here the Bfields are getting updated.

Reply:
oh never mind. I just read the PR description!!